### ISSUE TITLE
The unidirectional edge getOrigin / getDestination functions should check validity

### ISSUE THREAD

**Issue by nrabinowitz** [uber/h3#73]
At present, the functions `getOriginH3IndexFromUnidirectionalEdge` and `getDestinationH3IndexFromUnidirectionalEdge` assume a valid edge. This leads to some weird behavior when non-edges are passed in:
```
getOriginH3IndexFromUnidirectionalEdge('891ea6d6533ffff'); // '891ea6d6533ffff'
getOriginH3IndexFromUnidirectionalEdge(0); // '800000000000000'
```
I don't think we need to call `h3UnidirectionalEdgeIsValid`, since it might(?) have more performance implications, but perhaps adding a bit check like 
```
if (H3_GET_MODE(edge) != H3_UNIEDGE_MODE) {
  return H3_INVALID_INDEX;
}
```
would catch some common bad cases?

**Comment by isaachier** [comment]
What are the quotes?

**Comment by isaachier** [comment]
Anyway, I think throughout the library, you should be using better error checking where possible. As of now, my biggest concern is still `calloc` returning `NULL` when OOM occurs. If you are really concerned about performance, you can always follow this classic, ugly C pattern:

```
int func(int argc, const char* argv[])
{
#ifndef NDEBUG  // If not not debugging... AKA in debug mode
    const int returnCode = checkTheAssumptions(args);
    if (returnCode != 0) {
        return returnCode;
    }
#endif  // NDEBUG
...
    return 0;
}
```

**Comment by nrabinowitz** [comment]
> What are the quotes?

Sorry, I'm working on the JS bindings, so this was copied from there. It still holds in C.

**Comment by nrabinowitz** [comment]
Fixed in #74
